packages/editor/src/schema.ts

```diff
@@ -0,0 +1,371 @@
+import type { Attrs, DOMOutputSpec, Mark, MarkSpec, NodeSpec, Node as PMNode } from 'prosemirror-model';
+
+export declare type EditorMarks =
+  | 'link'
+  | 'em'
+  | 'strong'
+  | 'code'
+  | 'underline'
+  | 'strikethrough'
+  | 'subscript'
+  | 'superscript';
+// | 'style';
+
+export declare type EditorNodes =
+  | 'doc'
+  | 'paragraph'
+  | 'blockquote'
+  | 'horizontalRule'
+  | 'heading'
+  | 'codeBlock'
+  | 'text'
+  | 'image'
+  | 'hardBreak'
+  | 'listItem'
+  | 'orderedList'
+  | 'bulletList';
+
+const SLOT = 0; // https://prosemirror.net/docs/guide/#schema.serialization_and_parsing
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export const isEmpty = (obj: Object): boolean => Object.keys(obj).length === 0;
```

Comment:
Maybe we can use `(obj: Record<string, unknown>)` instead of `(obj: Object)` here?